feat-projectcompo-migrationFlow (#1612)
bonitasoft/bonita-doc: modules/ROOT/pages/upgradestudio.adoc

Maintenance versions are cross-compatible, meaning that it is not required that your Studio version matches the target runtime version.

For example, your production runtime can be in 2021.1-0408 and your Studio in 2021.1-0218

== Upgrading your project to the project composition

Version 2021.2 introduces major changes to the way the Studio manages projects. When importing or cloning your project in this version or any later version, it will be transformed into a Maven project. All your project dependencies (connectors, libraries, actor filters...) will be managed with Maven.

*_BOS Import_*

In the BOS import wizard, an additional step has been added. The Studio is going to scan your project dependencies and try to find them on the configured Maven repositories (by default Maven Central is configured).

> If the dependencies are found then they will be used as remote dependencies (not committed in your project) otherwise they will be added to the project local repository store and will be committed to your project. 

Reviewer: ```suggestion
If the dependencies are found then they will be used as remote dependencies (not copied in your project) otherwise they will be copied in the project and will be committed with your project sources. 
```

In both cases, dependencies that are in the project but are not used will be added in the upgraded project unless you add them manually (by ticking the corresponding checkboxes).

You can also remove the other dependencies (at your own risk).

*_Project cloning_*

> If your project is cloned through GIT, there isn't an extra step in the wizard. The migration is applied to the default branch or the branch selected in the GIT clone wizard.

Reviewer: ```suggestion
If your project is cloned through GIT, there isn't an extra step in the wizard. The migration is automatically applied to the checkouted branch. A `MIGRATION_NOTES.adoc` file is generated at the root of the project to details the major changes occuring at migration. It is up to you to keep it in the project or not (later migrations will be append to this file if it exists.)
```

The dependencies are also analyzed by the Studio but the result of the analysis is directly applied to your project.
The Studio is going to scan your project dependencies and try to find them on the configured Maven repositories (by default Maven Central is configured).

If the dependencies are found then they will be used as remote dependencies (not committed in your project) otherwise they will be added to the project local repository store and will be committed to your project. 

*_Upgrading the local repository dependencies to remote dependencies_*

> You replace the local repository dependencies with remote dependencies (if they are made available on a remote repository) by accessing the xref:managing-extension-studio.adoc["Extension" view]. In the "Other extensions" part, select the local repository dependencies and use the "Edit maven coordinates" button to use the corresponding Maven coordinates to get the remote dependency.

Reviewer: The proper way of doing it for now is to delete the local dep and re-add it using maven coordinates

> === Impact Extension creation

Reviewer: Not clear

> From 2021.2, it will no longer be possible to create connectors, actor filters, and themes from the Studio. You will have to use the corresponding xref:software-extensibility.doc[Maven archetypes].

Reviewer: ```suggestion
[CAUTION]
====
From 2021.2, it will no longer be possible to create connectors, actor filters, and themes from the Studio. You will have to use the corresponding xref:software-extensibility.doc[Maven archetypes].
====
```
